src/dev/i18n/extract_default_translations.test.js

```diff
@@ -17,29 +17,72 @@
  * under the License.
  */
 
-import { resolve } from 'path';
+import path from 'path';
 import fs from 'fs';
 import { promisify } from 'util';
+import JSON5 from 'json5';
+import { i18n } from '@kbn/i18n';
 
 import { extractDefaultTranslations } from './extract_default_translations';
 
 const readFileAsync = promisify(fs.readFile);
 const removeDirAsync = promisify(fs.rmdir);
 const unlinkAsync = promisify(fs.unlink);
 
-const PLUGIN_PATH = resolve(__dirname, '__fixtures__', 'test_plugin');
+const fixturesPath = path.resolve(__dirname, '__fixtures__', 'extract_default_translations');
+const pluginsPaths = [
+  path.join(fixturesPath, 'test_plugin_1'),
+  path.join(fixturesPath, 'test_plugin_2'),
+  path.join(fixturesPath, 'test_plugin_3'),
+];
 
 describe('dev/i18n/extract_default_translations', () => {
+  it('extracts messages to en.json', async () => {
+    await extractDefaultTranslations(pluginsPaths[0]);
+
+    const extractedJSONBuffer = await readFileAsync(
+      path.join(pluginsPaths[0], 'translations', 'en.json')
+    );
+
+    await unlinkAsync(path.join(pluginsPaths[0], 'translations', 'en.json'));
+    await removeDirAsync(path.join(pluginsPaths[0], 'translations'));
+
+    const expectedJSONFormats = JSON5.stringify(
+      { formats: i18n.formats },
+      { quote: `'`, space: 2 }
+    );
+
+    const expectedJSON = `${expectedJSONFormats.slice(0, -2)}
+  'plugin_1.id_1': 'Message 1',
+  'plugin_1.id_2': 'Message 2', // Message context
+  'plugin_1.id_3': 'Message 3',
+  'plugin_1.id_4': 'Message 4',
+  'plugin_1.id_5': 'Message 5',
+  'plugin_1.id_6': 'Message 6',
+  'plugin_1.id_7': 'Message 7',
+}
+`;
+
+    expect(extractedJSONBuffer.toString()).toEqual(expectedJSON);
```

Comment:
__nit:__ probably use `expect(extractedJSONBuffer.toString()).toMatchSnapshot()` instead

Also in other tests below you can use snapshot feature more frequently ;)